packages/block/src/block.ts

```diff
@@ -978,6 +999,29 @@ export class Block {
       ...withdrawalsArr,
       parentBeaconBlockRoot: header.parentBeaconBlockRoot,
       executionWitness: this.executionWitness,
+
+      // lets add the  request fields first and then iterate over requests to fill them up
+      depositRequests: this.common.isActivatedEIP(6110) ? [] : undefined,
+      withdrawalRequests: this.common.isActivatedEIP(7002) ? [] : undefined,
+    }
+
+    if (this.requests !== undefined) {
+      for (const request of this.requests) {
```

Comment:
These requests should be ordered. I'm not sure if a `switch / case` structure is the "nicest" way to handle this :thinking:

Reply:
the sorting is verified as part of block validations so we don't need to worry and keep things simple here